src/app/portlets/dot-starter/dot-starter-resolver.service.ts

```diff
@@ -1,20 +1,46 @@
-import { Observable, of } from 'rxjs';
+import { Observable } from 'rxjs';
 import { Injectable } from '@angular/core';
 import { Resolve } from '@angular/router';
-import { LoginService } from 'dotcms-js';
+import {
+    DotCurrentUserService,
+    PermissionsType,
+    UserPermissions
+} from '@services/dot-current-user/dot-current-user.service';
+import { DotCurrentUser, DotPermissionsType } from '@models/dot-current-user/dot-current-user';
+import { map, mergeMap } from 'rxjs/operators';
 
 /**
- * Returns username logged
+ * Returns user's data and permissions
  *
  * @export
  * @class DotStarterResolver
- * @implements {Resolve<Observable<string>>}
+ * @implements {Resolve<Observable<[DotCurrentUser, DotPermissionsType]>>}
  */
 @Injectable()
-export class DotStarterResolver implements Resolve<Observable<string>> {
-    constructor(private loginService: LoginService) {}
+export class DotStarterResolver
+    implements Resolve<Observable<[DotCurrentUser, DotPermissionsType]>> {
+    constructor(private dotCurrentUserService: DotCurrentUserService) {}
 
-    resolve(): Observable<string> {
-        return of(this.loginService.auth.user.firstName);
+    resolve(): Observable<[DotCurrentUser, DotPermissionsType]> {
+        return this.dotCurrentUserService.getCurrentUser().pipe(
+            map((user: DotCurrentUser) => {
+                return this.dotCurrentUserService
+                    .getUserPermissions(
+                        user.userId,
+                        [UserPermissions.WRITE],
+                        [
+                            PermissionsType.HTMLPAGES,
+                            PermissionsType.TEMPLATES,
+                            PermissionsType.CONTENTLETS
+                        ]
+                    )
+                    .pipe(
+                        map((permissionsType: DotPermissionsType[]) => {
+                            return [user, permissionsType[0]];
+                        })
+                    );
+            }),
+            mergeMap((data: Observable<[DotCurrentUser, DotPermissionsType]>) => data)
```

Comment:
You don't need the map just the `mergeMap`, and maybe you should a `switchMap` which will always return the latest request.